lib/extractor/layer.ts

```diff
@@ -53,6 +56,73 @@ export async function extractDockerArchive(
   });
 }
 
+export async function extractOciArchive(
+  ociArchiveFilesystemPath: string,
+  extractActions: ExtractAction[],
+): Promise<{
+  layers: ExtractedLayers[];
+  manifest: OciArchiveManifest;
+}> {
+  return new Promise((resolve, reject) => {
+    const tarExtractor: Extract = extract();
+
+    const layers: Record<string, ExtractedLayers> = {};
+    const manifests: Record<string, OciArchiveManifest> = {};
+    let imageIndex: OciImageIndex | undefined;
+
+    tarExtractor.on("entry", async (header, stream, next) => {
+      if (header.type === "file") {
+        if (isImageIndexFile(header.name)) {
+          imageIndex = await streamToJson<OciImageIndex>(stream);
+        } else {
+          const jsonStream = new PassThrough();
+          const layerStream = new PassThrough();
+          stream.pipe(jsonStream);
+          stream.pipe(layerStream);
+
+          const promises = [
+            streamToJson(jsonStream).catch(() => undefined),
+            extractImageLayer(layerStream, extractActions).catch(
+              () => undefined,
+            ),
+          ];
+          const [manifest, layer] = await Promise.all(promises);
+
+          // header format is /blobs/hash_name/hash_value
```

Comment:
Actually, any chance to assert that this is true and bail early if not? E.g.
```js
const headerParts = header.name.split("/");
if (!(headerParts.length === 3 && headerParts[0] === 'blobs')) {
    return; // or break; or however we need to continue
}
```

Is there a chance we could get an error if we received a file that is neither a layer nor a manifest?